### DISCUSSION TITLE
Bot Replies to Commands in Threads

### DISCUSSION THREAD

**Discussion by FreestRolexes** [Rapptz/discord.py#7328] [Q&A]
Per the title, I'm looking for help getting my bot to reply to commands from within threads.  

I've updated `discord.py` to v1.7.3, which (if I'm understanding right) should implement the Discord API v9 and support threads.  But I'm not seeing any messages from within a thread with my `@bot.listen('on_message')` wrapped function, in addition to any of my command handlers.  My bot is still able to respond to commands from DMs or from within normal channels on the server it is attached to.  Is there something obvious I'm missing or additional steps I need to take to support threads?

I apologize if this isn't the proper place to ask technical questions.

**Comment by XuaTheGrate** [comment]
> I've updated discord.py to v1.7.3, which (if I'm understanding right) should implement the Discord API v9 and support threads.

This is incorrect. Thread support is added in v2.0, which has not been released yet. You can install it with github:
```
pip install -U git+https://github.com/Rapptz/discord.py
```

**Reply by FreestRolexes** [reply]
Ah!  That makes a ton of sense.  Thanks for the help (and providing the git command)!  v2.0 works like a charm.

**Comment by Kev-in123** [comment]
Discord.py v1.7.3 was released before threads.

**Reply by Phxntxm** [reply]
That's nice
